### ISSUE TITLE
CodeAction should be resolved only when user click on QuickFix

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/intellij-quarkus#785]
Qute / Quarkus LS provides some code action with resolve support, generally it is code action for Java file which is done with resolve.

It is very important to resolve code action on click on the quick fix and not when diagnostic is published. Today code action for Java file are resolved (we compute text edit for all code actions by using PsiJavaFile which can takes some times and are not very good for performance).

In other words the compute of text edit by using PsiJavaFile must be done on click of the quick fix and not when the editor is validated. If we have several code actions, today all code actions are resolved while the validation process, we compute them although user don't use it.
